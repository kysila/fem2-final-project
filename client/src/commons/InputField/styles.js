import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles(() => ({
  formControl: {
    width: '100%',
    marginTop: '12px',
  },
  label: {
    zIndex: 1,
    transform: 'translate(14px, 12px) scale(1)',
    pointerEvents: 'none',
    fontSize: '14px',
  },
  input: {
    width: '100%',
    padding: '10px 14px',
    borderRadius: '3px',
  },
}));
