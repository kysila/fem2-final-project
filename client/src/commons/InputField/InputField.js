import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { FormControl, InputLabel, OutlinedInput } from '@material-ui/core';


import VisibilityIcon from '@material-ui/icons/Visibility';
import VisibilityOffIcon from '@material-ui/icons/VisibilityOff';
import { ArrowTooltip } from '../Tooltip/Tooltip';

import { useStyles } from './styles';

function InputField(props) {
  const classes = useStyles();
  const {
    id,
    value,
    onChange,
    onBlur,
    onFocus,
    labelWidth,
    label,
    type,
    variant,
    FormControlProps,
    LabelProps,
    InputProps,
    TooltipProps,
    error,
  } = props;

  const [passwordHidden, setPasswordVisible] = useState(true);
  const PasswordIcon = passwordHidden ? VisibilityIcon : VisibilityOffIcon;

  return (
    <FormControl className={classes.formControl} variant={variant} {...FormControlProps}>
      <InputLabel classes={{ root: classes.label }} htmlFor={id} {...LabelProps}>
        {label}
      </InputLabel>
      <ArrowTooltip title={error} {...TooltipProps}>
        <OutlinedInput
          id={id}
          type={type === 'password' && passwordHidden ? type : 'text'}
          classes={{
            input: classes.input,
          }}
          error={Boolean(error)}
          value={value}
          onChange={onChange}
          onBlur={onBlur}
          onFocus={onFocus}
          labelWidth={labelWidth}
          fullWidth
          {
          ...(type === 'password' && value ? {
            endAdornment: (
              <PasswordIcon
                style={{ cursor: 'pointer', userSelect: 'none' }}
                onTouchStart={() => setPasswordVisible(false)}
                onMouseDown={() => setPasswordVisible(false)}
                onTouchEnd={() => setPasswordVisible(true)}
                onMouseUp={() => setPasswordVisible(true)}
              />
            ),
          } : {})
        }
          {...InputProps}
        />
      </ArrowTooltip>
    </FormControl>
  );
}

InputField.propTypes = {
  type: PropTypes.string,
  variant: PropTypes.string,
  labelWidth: PropTypes.number,
  value: PropTypes.string,
  label: PropTypes.string,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,
  error: PropTypes.node,
  FormControlProps: PropTypes.shape(FormControl.propTypes),
  LabelProps: PropTypes.shape(InputLabel.propTypes),
  InputProps: PropTypes.shape(OutlinedInput.propTypes),
  TooltipProps: PropTypes.shape(ArrowTooltip.propTypes),
};

InputField.defaultProps = {
  type: 'text',
  variant: 'outlined',
  labelWidth: 0,
  label: '',
  value: '',
  onChange: () => null,
  onBlur: () => null,
  onFocus: () => null,
  error: null,
  FormControlProps: {},
  LabelProps: {},
  InputProps: {},
  TooltipProps: {},
};

export {
  InputField,
};
