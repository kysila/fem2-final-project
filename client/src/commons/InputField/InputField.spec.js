/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { InputField } from './InputField';

const propsText = {
  id: 'test-id',
  type: 'text',
  label: 'Email',
  labelWidth: 40,
  value: '123',
};

const propsPassword = {
  id: 'test-id',
  type: 'password',
  label: 'Password',
  labelWidth: 70,
  value: 'qwerty123',
};

describe('InputField component', () => {
  it('InputField render text', () => {
    const rendered = shallow(
      <InputField
        {...propsText}
      />,
    );

    expect(rendered).toMatchSnapshot();
  });

  it('InputField render password', () => {
    const rendered = shallow(
      <InputField
        {...propsPassword}
      />,
    );

    expect(rendered).toMatchSnapshot();
  });

});
