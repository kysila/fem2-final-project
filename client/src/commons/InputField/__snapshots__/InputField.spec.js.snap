// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField component InputField render password 1`] = `
<WithStyles(ForwardRef(FormControl))
  className="makeStyles-formControl-1"
  variant="outlined"
>
  <WithStyles(ForwardRef(InputLabel))
    classes={
      Object {
        "root": "makeStyles-label-2",
      }
    }
    htmlFor="test-id"
  >
    Password
  </WithStyles(ForwardRef(InputLabel))>
  <ArrowTooltip
    title={null}
  >
    <WithStyles(ForwardRef(OutlinedInput))
      classes={
        Object {
          "input": "makeStyles-input-3",
        }
      }
      endAdornment={
        <Memo(VisibilityIcon)
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "userSelect": "none",
            }
          }
        />
      }
      error={false}
      fullWidth={true}
      id="test-id"
      labelWidth={70}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="password"
      value="qwerty123"
    />
  </ArrowTooltip>
</WithStyles(ForwardRef(FormControl))>
`;

exports[`InputField component InputField render text 1`] = `
<WithStyles(ForwardRef(FormControl))
  className="makeStyles-formControl-1"
  variant="outlined"
>
  <WithStyles(ForwardRef(InputLabel))
    classes={
      Object {
        "root": "makeStyles-label-2",
      }
    }
    htmlFor="test-id"
  >
    Email
  </WithStyles(ForwardRef(InputLabel))>
  <ArrowTooltip
    title={null}
  >
    <WithStyles(ForwardRef(OutlinedInput))
      classes={
        Object {
          "input": "makeStyles-input-3",
        }
      }
      error={false}
      fullWidth={true}
      id="test-id"
      labelWidth={40}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value="123"
    />
  </ArrowTooltip>
</WithStyles(ForwardRef(FormControl))>
`;
